node/node_handler.go

```diff
@@ -264,6 +264,16 @@ func (node *Node) VerifyNewBlock(newBlock *types.Block) error {
 			"my shard ID", node.Blockchain().ShardID(),
 			"new block's shard ID", newBlock.ShardID())
 	}
```

Comment:
This VerifyNewBlock is only called for consensus block verification, but not for block sync verification. We should consolidate it into a single func called by all places that requires block verification. Please add a todo.

Reply:
I first want to put it into verifyHeader, but verifyShardState need beaconchain as parameter, so I put it outside. will add a todo